src/components/reserve/api.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import { get, map } from 'lodash'
+import API from '../../lib/API/api'
+import GoodWallet from '../../lib/wallet/GoodWallet'
+
+type GoodDollarStats = {
+  price: number,
+  priceChange?: number,
+  priceChangePercentage?: number,
+  priceIncreasing?: boolean,
+  priceDecreasing?: boolean,
+}
+
+class ReserveAPI {
+  http = null
+
+  wallet = null
+
+  constructor(httpClient, wallet) {
+    this.http = httpClient
+    this.wallet = wallet
+  }
+
+  async getGoodDollarPrice(withStats: boolean = true): Promise<GoodDollarStats> {
+    const { http, wallet } = this
+    const price = await wallet.getReservePriceDAI()
+    const tokenStats: GoodDollarStats = { price }
+
+    if (false !== withStats) {
+      const [currentPrice, previousPrice] = await http
+        .getGoodDollarReservePrice()
+        .then(gqlResponse => get(gqlResponse, 'data.reserveHistories', []))
+        .then(priceHistory => map(priceHistory, 'openPriceDAI'))
+
+      if (currentPrice > 0 && previousPrice > 0) {
```

Comment:
the last prices from reserve can be > 1 days if no one bought from reserve.
the graph will only update if someone bought/sold on reserve.